### ISSUE TITLE
Extracting and analyzing "single units" after training

### ISSUE THREAD

**Issue by claudio1212** [neurogym/neurogym#270]
Hello everyone, I manage to train a model in performing a modified version of a GoNoGo task, following the updated notebook "Understanding NeuroGym Tasks", that I have to say is very well done. So after following step by step that tutorial, and applying some minor tweaks to adjust the task to my needs, I was wondering what are the next steps that I would need to take in order to analyze the units of the trained model, as for example was done in the paper that is cited with the toolbox. I saw that there is another github section called "ngym_usage" but I was not able to understand the use of those functions and apply them to the example of the notebook I used.

**Comment by cantordust** [comment]
Apropos, there is preliminary support for extracting neuron traces in the `dev` branch. You can look at [this notebook](https://github.com/neurogym/neurogym/blob/dev/docs/examples/neuron_activation_traces.ipynb).

This is still in a beta state, so don't rely on it too much, but it would be great to let me know if it does anything useful for you (and, of course, if you find bugs, please open an issue).

**Comment by claudio1212** [comment]
Thank you both for you replies, I'll try to answer to both: At the moment, i trained a RecurrentPPO model using the following commands (they are an example, I tried many different parameters for the model): from neurogym.wrappers import pass_reward
env = PrecisionGripTask()
env = pass_reward.PassReward(env)
from sb3_contrib import RecurrentPPO
from stable_baselines3.common.vec_env import DummyVecEnv
policy_kwargs = dict(net_arch=[256, 256])

env = DummyVecEnv([lambda: env])

model = RecurrentPPO(
    policy="MlpLstmPolicy",
     learning_rate=3e-4,  
    #Tasso di apprendimento dell’ottimizzatore. Valori più bassi rendono l’apprendimento più lento ma stabile; v
    # valori più alti possono accelerare ma rischiano di rendere instabile la convergenza.
    gamma=0.99,            
#Determina quanto l’agente tiene conto delle ricompense future.
#1.0: considera solo la somma totale delle ricompense future
#Valori più bassi: privilegia ricompense immediate 
    n_steps=512,            
    
    #Numero di step che l’agente esegue prima di aggiornare la policy (lunghezza dei rollout).
#Valori bassi: aggiornamenti più frequenti, ma con maggiore varianza
#Valori alti: aggiornamenti più stabili, ma meno frequenti
#se c'è variabilità nelle ricompense, si può aumentare 


    ent_coef=0.05,       
    env=env,
    vf_coef=0.5,          # bilanciamento loss value
    max_grad_norm=0.5,    # stabilità nei gradienti
    verbose=1,
    policy_kwargs=policy_kwargs
)
model.learn(total_timesteps=500000, log_interval=100).  

I'd like to replicate some kind of simple analyses such as computing the mean firing rate of the "population of units" that were identified, in different periods of the employed task,  but also to perform some dimensionality reduction analyses such as PCA, to see the neural dynamics in time throughout the course of the task. As I mentioned above, the behavioral task is a slightly modified version of the ones available in the example notebook, and has different time periods in which different events happen (cue onset, object presentation ecc.)

**Comment by claudio1212** [comment]
I'll try the notebook as soon as I can and let you know, thank you so much for your help!
